src/main/java/core/basesyntax/Application.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+public class Application {
+    private static final int arrayLenght = 6;
+    private static FigureSupplier figureSupplier = new FigureSupplier();
+
+    public static void main(String[] args) {
+        Figure[] figureArray = new Figure[arrayLenght];
+
+        for (int i = 0; i < arrayLenght / 2; i++) {
```

Comment:
Don't use a static field for 'figureSupplier', instantiate it in the 'main' method to avoid static methods and fields as much as possible.